shm/player.hpp

```diff
@@ -0,0 +1,36 @@
+#pragma once
+#include <iostream>
+#include <memory>
+#include <vector>
+#include "ship.hpp"
+#include "cargo.hpp"
+
+class Player {
+private:
+    std::shared_ptr<Ship> ship_;
+    size_t money_;
+    size_t availableSpace_;
+
+    size_t calculateAvSpace(size_t capacity, size_t cargo) {
+        auto capacity = ship_->getCapacity();
+        auto cargo = ship_->getCargo();
+        return capacity - cargo;
+    }
```

Comment:
This function should be called in C'tor instead of passing available space. Also, keep in mind that you need to recalculate it every time you add/ remove cargo.